control/HTTP.php

```diff
@@ -371,8 +371,10 @@ public static function add_cache_headers($body = null) {
 		if(headers_sent() && !$body) {
 			return;
 		}
-		// Skip already assigned cache-control headers
+
+		// Warn if already assigned cache-control headers
 		if ($body && $body->getHeader('Cache-Control')) {
+			trigger_error("Cache-Control header has already been set", E_USER_WARNING);
```

Comment:
In addition, we should say something like "Please use HTTPCacheControl::singleton() methods to more safely customise your cache-control headers."

Reply:
How about 

```php
trigger_error(
	"Cache-Control header has already been set. "
	. "Please use HTTPCacheControl API to set caching options instead.",
	E_USER_WARNING
);
```